cri-o/cri-o
# Add bandwidth limiting support

**- What I did**
Added support for bandwidth limiting CNI plugins (it already exists within dockershim in kubelet). Default burst values have been copied from Kubelet (although those are `int` instead of `int64`, but I would argue that `int` is not enough on 10Gbit networks).

Please note that this  PR depends on merging https://github.com/cri-o/ocicni/pull/28 - after which this code needs to be revendored! Until then it will not compile.

**- How to verify it**
Create a pod with these annotations:
```
      annotations:
        kubernetes.io/egress-bandwidth: 20M
        kubernetes.io/ingress-bandwidth: 20M
```

and when you launch it there should be `qdisc` set:
```
# tc qdisc show
...
qdisc tbf 1: dev cali9d94ec92eeb root refcnt 2 rate 20Mbit burst 171798690b limit 62499b 
qdisc tbf 1: dev 76ae root refcnt 2 rate 20Mbit burst 171798690b limit 62499b 
...
```
**- Description for the changelog**
Added bandwidth CNI limiting support
